paasta_tools/cli/cmds/spark_run.py

```diff
@@ -536,8 +542,37 @@ def get_spark_app_name(original_docker_cmd: Union[Any, str, List[str]]) -> str:
 
     if spark_app_name is None:
         spark_app_name = "paasta_spark_run"
-
     spark_app_name += f"_{get_username()}"
+    if autogen != '-1':
+        document = f"""
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    spark: exec
+spec:
+  affinity:
+    podAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: {autogen}
```

Comment:
Is the command line value passed to spark-run also the value for the weight? It doesn't seem to be part of the flag description. I would prefer using a boolean flag + number flag eg. `--enable-k8s-autogen --k8s-affinity-weight` or something along the line. If possible the weight should have a default value that's somewhat sensible as well.

Reply:
yeah as it stands it's `--auto-gen 95` to supply both the weight and the automatically generate the file, because otherwise it's just a superlong argument someone to supply

Reply:
cool I'll change the default, and also put a better description on `--auto-gen`

Reply:
made this into a f string now

Reply:
also updated the description

Reply:
I think by default we would want to enable autogen (with some default optimal weight) as we would want to prefer pod affinity in all cases.
Makes migration simpler and less arguments to play around or learn in future.  Thoughts?

Reply:
yeah as above with default value make sense